pkg/cmd/dcos_cluster_list.go

```diff
@@ -0,0 +1,122 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/dcos/dcos-cli/pkg/cli"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterList lists the clusters.
+func newCmdClusterList(ctx *cli.Context) *cobra.Command {
+	var onlyAttached bool
+	var jsonOutput bool
+	cmd := &cobra.Command{
+		Use:  "list",
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clusters, err := ctx.Clusters()
+			if err != nil {
+				return err
+			}
+
+			currentCluster, err := ctx.Cluster()
+			if err != nil {
+				return err
+			}
+			currentClusterID := currentCluster.ID()
+
+			// clustersData is composed of clusters containing its name, id, status, version, and url.
+			infos := []clusterInfo{}
+			var mu sync.Mutex
+
+			var wg sync.WaitGroup
+			for _, cluster := range clusters {
+				// Add should execute before the statement creating the goroutine.
+				wg.Add(1)
+				go func(cluster *cli.Cluster) {
+					defer wg.Done()
+
+					var info clusterInfo
+					info.Name = cluster.Name()
+					info.ID = cluster.ID()
+					if info.ID == currentClusterID {
+						info.Attached = true
+						if !jsonOutput {
+							info.Name += "*"
```

Comment:
As discussed yesterday, could this be in an empty column at the beginning instead?